aten/src/ATen/native/SummaryOps.cpp

```diff
@@ -64,4 +65,56 @@ _bincount_cpu(const Tensor& self, const Tensor& weights, int64_t minlength) {
   });
 }
 
+Tensor& histc_out(Tensor& hist, const Tensor &self, int64_t nbins, Scalar minvalue, Scalar maxvalue) {
+  if (nbins <= 0) {
+    AT_ERROR("bins must be > 0");
+  }
+
+  auto tensor = self.contiguous();
+  hist.resize_({nbins});
+  hist.zero_();
+
+  AT_DISPATCH_FLOATING_TYPES(tensor.scalar_type(), "histc_cpu", [&]() -> void {
+    scalar_t minval;
+    scalar_t maxval;
+    int64_t *h_data;
+    scalar_t *tensor_data;
+
+    minval = minvalue.to<scalar_t>();
+    maxval = maxvalue.to<scalar_t>();
+
+    if (minval == maxval) {
+      minval = tensor.min().item().to<scalar_t>();
+      maxval = tensor.max().item().to<scalar_t>();
+    }
+    if (minval == maxval) {
+      minval = minval - 1;
+      maxval = maxval + 1;
+    }
+
+    TORCH_CHECK(!(std::isinf(minval) || std::isinf(maxval) || std::isnan(minval) || std::isnan(maxval)), "range of [", minval, ", ", maxval, "] is not finite");
+    TORCH_CHECK(minval < maxval, "max must be larger than min");
+
+    h_data = hist.data_ptr<int64_t>();
+    tensor_data = tensor.data_ptr<scalar_t>();
+
+    at::parallel_for(0, tensor.numel(), 1, [&](int64_t i_begin, int64_t i_end) {
```

Comment:
Consider using unary TensorIterator loop instead of casting input to contiguous and using at::parallel

Reply:
thank you. But I have a question, Can TensorIterator handle the `input`&`output` tensor have not same size's situation? because I can't found the example.

Reply:
Just do one input and no outputs for tensor iterator.  It maybe faster than doing `contiguous` copy.

Reply:
done